xrootd/protocol/dirlist/dirlist.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package dirlist contains the structures describing request and response
+// for dirlist request used to obtain the contents of a directory.
+package dirlist // import "go-hep.org/x/hep/xrootd/protocol/dirlist"
+
+// RequestID is the id of the request, it is sent as part of message.
+// See xrootd protocol specification for details: http://xrootd.org/doc/dev45/XRdv310.pdf, 2.3 Client Request Format.
+const RequestID uint16 = 3004
+
+// Response is a response for the dirlist request,
+// which contains a byte array containing encoded response.
+// The format (if stat information is supported by the server) is:
+// ".\n"
```

Comment:
please indent these by a couple of spaces so they are properly formatted in `godoc`:

```go
// The format (if stat information is supported by the server) is:
//     ".\n"
//     "0 0 0 0\n"
//     "dirname\n"
//     "id size flags modtime\n"
//     ...
//     0
// 
// In case that the server doesn't support returning the stat information, the format is:
//     "dirname\n"
//     ...
//     0
// 
// See xrootd protocol specification, page 45 for further details.
```

Reply:
that said, it's perhaps a bit too low-level (or nitty-gritty details.)

Ideally, the `Response` should come out of the pipe with a slice of `EntryStat`...
(that should probably be done by `func (resp *Response) UnmarshalXrd(data []byte) error`)